packages/shared/src/hooks/usePostModalNavigation.ts

```diff
@@ -111,22 +119,20 @@ export const usePostModalNavigation = ({
       if (post) {
         const postId = post.slug || post.id;
 
-        await router.push(
-          {
-            pathname: basePathname,
-            query: {
-              ...router.query,
-              pmid: postId,
-              pmp: basePathname,
-              pmap: baseAsPath,
-              pmcid: feedName,
-            },
-          },
-          `${webappUrl}posts/${postId}`,
-          {
-            scroll: false,
-          },
+        const newPathname = getPathnameWithQuery(
+          basePathname,
+          objectToQueryParams({
+            ...router.query,
+            pmid: postId,
+            pmp: basePathname,
+            pmap: baseAsPath,
+            pmcid: feedName,
+          }),
         );
```

Comment:
Just changed to pass `string` to be compatible with in memory router as well. Luckly we have helpers already 👌